libs/coin-modules/coin-aptos/src/bridge/buildTransaction.ts

```diff
@@ -11,18 +11,51 @@ const buildTransaction = async (
   transaction: Transaction,
   aptosClient: AptosAPI,
 ): Promise<RawTransaction> => {
-  const txPayload = getPayload(transaction.recipient, transaction.amount);
+  const subAccount = findSubAccountById(account, transaction.subAccountId ?? "");
+
+  const txPayload = getPayload(subAccount, transaction);
   const txOptions = normalizeTransactionOptions(transaction.options);
   const tx = await aptosClient.generateTransaction(account.freshAddress, txPayload, txOptions);
 
   return tx;
 };
 
-const getPayload = (sendTo: string, amount: BigNumber): InputEntryFunctionData => {
+const getPayload = (
+  tokenAccount: TokenAccount | undefined,
+  transaction: Transaction,
+): InputEntryFunctionData => {
+  if (tokenAccount && isTokenAccount(tokenAccount)) {
+    const { tokenType } = tokenAccount.token;
+
+    if (!SUPPORTED_TOKEN_TYPES.includes(tokenType)) {
+      throw new Error(`Token type ${tokenType} not supported`);
```

Comment:
This is unnecessary logic, we can remove it

If there is another token type that needs to be supported in the future, we will need to update the app anyway to register it in another CAL

Also do we support all token types ?

Reply:
We don't support non-fungible assets like NFTs.